src/components/tutorials-sidebar/tutorial-progress-icon/tutorial-progress-icon.module.css

```diff
@@ -0,0 +1,28 @@
+.progressIcon {
+	width: 12px;
+	height: 12px;
+}
```

Comment:
🙋🏻 Purely out of my own curiosity, what led you to deciding on this approach? I personally probably would've done something like this, just based on my own habits:

```css
.iconComplete,
.iconInProgress,
.iconVisited {
  height: 12px;
  width: 12px;
}

.iconComplete {
  color: var(--token-color-palette-green-200);
}

.iconInProgress {
  color: var(--token-color-palette-blue-200);
}

.iconVisited {
  color: magenta;
}
```

OR I might have done something like this with variables and named the classes in a way that they could've been applied using the `classnames` package:

```css
.root {
  color: var(--icon-color);
  height: 12px;
  width: 12px;
}

.complete {
  --icon-color: var(--token-color-palette-green-200);
}

.in_progress {
  --icon-color: var(--token-color-palette-blue-200);
}

.visited {
  --icon-color: magenta;
}
```

No action needed here, just riffing 🎸

Reply:
I think there are a couple things that make me lean towards `compose` in this case:

- **One rule block per selector**
    - with compose, I can write exactly one rule block for `.iconComplete`, and it forms a complete picture of the styles being applied. The use of `compose` means the connection between `.iconComplete` and `.progressIcon` is clear from looking at that one rule block.
    - Where in the first example, unless you can see the whole file at once, neither of the two rule blocks that apply to `.iconComplete` necessarily indicate that there's another place you need to look in order to get a complete picture of the relevant styles.

That's obviously not a big distinguishing factor in this case, but I dig "one rule block per selector" as a more general rule so that I don't ever have to worry "am i missing some other relevant styles somewhere else in the file". There's a more relevant example in the Notion page [Code Style & Practices > CSS Modules](https://www.notion.so/hashicorp/CSS-Modules-753f2647f7aa4a598e28990d9f0d05a8), under the `Group styles by element, not by state or media type` point under `Recommended Practices`.

- **Cleaner markup**
    - this is a really minor one compared to the above IMO... but in cases where either a single `compose` or the CSS custom props in the second example will work for a use case, I'd lean very slightly towards `compose` to keep the markup and style mapping simpler (exactly one associated `className` to look at).
    - I also find it _slightly_ easier to trace `compose` than CSS custom props, since the former is very explicit about where to look

That being said, I also really dig the use of CSS custom props for anything even just slightly more complex (eg, maybe the `--size` and `--icon-color` varies!)... so would be very on the fence deciding between the second example and `compose` as used in this PR 👍 Also `compose` seems to have rules around using it with nested selectors that I can never quite remember, so generally very on board with the CSS custom props approach 💯

Reply:
Awesome! 👏🏻 Thank you for all the detail. This makes total sense. 😊